llvm/test/CodeGen/RISCV/rvv/splat-vectors.ll

```diff
@@ -105,3 +105,43 @@ define <vscale x 4 x float> @splat_idx_nxv4f32(<vscale x 4 x float> %v, i64 %idx
   %splat = shufflevector <vscale x 4 x float> %ins, <vscale x 4 x float> poison, <vscale x 4 x i32> zeroinitializer
   ret <vscale x 4 x float> %splat
 }
+
+define <vscale x 4 x float> @splat_idx_constant_nxv8f32(<vscale x 8 x float> %v) {
```

Comment:
I had a bunch of comments on your testing, and finally decided it would be faster to just precommit them myself.  See e19bc7681.  You had a bunch of missing cases - both positive and negative.